### PR TITLE
Device control app data store

### PR DESCRIPTION
**Pull request by codedeal**

### CODE REVIEW

File: DeviceControlApp.Core/ViewModel/GpsStatusViewModel.cs [hunk 1]
```diff
@@ -24,13 +31,15 @@ public string Message
...
-        public GpsStatusViewModel(IPageService pageService,IGpsSensorService gpsSensorService,IFactory factory)
+        public GpsStatusViewModel(IPageService pageService,IGpsSensorService gpsSensorService,IFactory factory,IDataStore dataStore)
         {
              
             _factory = factory;
             _pageService = pageService;
             _gpsSensorService = gpsSensorService;
              CheckLocationServiceisEnabled();
+            _dataStore = dataStore;
+            Name = _dataStore.Get<string>("Name");
```
[the remark this answers is not in the document]
Comment (codedeal) [reply DeviceControlApp.Core/ViewModel/GpsStatusViewModel.cs]: HomePageWelcomeName is it good

File: DeviceControlApp.Core/ServiceImpln/DataStore.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                return (T)_map[key];
+            }
+            else
+            {
+               
+                throw new Exception("obj not found");
+            }
+        }
+
+        public bool IsEmpty()
```
[the remark this answers is not in the document]
Comment (codedeal) [reply DeviceControlApp.Core/ServiceImpln/DataStore.cs]: Done

File: DeviceControlApp.Core/ViewModel/GpsStatusViewModel.cs [hunk 3]
```diff
@@ -12,7 +12,14 @@ public class GpsStatusViewModel:BaseViewModel
         public IPageService _pageService;
         public ILocationService _locationService;
         public IGpsSensorService _gpsSensorService;
+        public IDataStore _dataStore;
 
+        public string Name
+        {
+            get;
+            set;
```
[the remark this answers is not in the document]
Comment (codedeal) [reply DeviceControlApp.Core/ViewModel/GpsStatusViewModel.cs]: done

File: DeviceControlApp.Core/ViewModel/GpsStatusViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nachiappannk) [reply DeviceControlApp.Core/ViewModel/GpsStatusViewModel.cs]: How about LoggedInUserName?

**Pull request merged**